src/key.rs

```diff
@@ -511,27 +639,42 @@ impl Ord for PublicKey {
 }
 
 /// Opaque data structure that holds a keypair consisting of a secret and a public key.
-// Should secrets implement Copy: https://github.com/rust-bitcoin/rust-secp256k1/issues/363
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// # #[cfg(feature="rand")] {
+/// use secp256k1::{KeyPair, Secp256k1};
+/// use secp256k1::rand::rngs::OsRng;
+///
+/// let secp = Secp256k1::new();
+/// let mut rng = OsRng::new().expect("OsRng");
```

Comment:
Why not `thread_rng`?

Reply:
I'm going to remove all the `OsRng`s and use `thread_rng` througout. Will do as a separate patch.